File: Code/Websites/DanpheEMR/Services/OnlinePayment/DTO/FewaPay/FewaPayTransactionLogDTO.cs
using System;

namespace DanpheEMR.Services.OnlinePayment.DTO.FewaPay
{
    /// <summary>
    /// FewaPayTransactionLogDTO is to transfer data from the client to save the FewaPay Transaction logs
    /// </summary>
    public class FewaPayTransactionLogDTO
    {
        /// <summary>
        /// PatientId Refers to the Patient for which transaction is being done
        /// </summary>
        public int PatientId { get; set; }
        /// <summary>
        /// ModuleName refers to the module from where transaction is being done
        /// </summary>
        public string ModuleName { get; set; }
        /// <summary>
        /// TransactionId created by respective Payment Service Providers after successful payment
        /// </summary>
        public string TransactionId { get; set; }
        /// <summary>
        /// TransactionType refers to the Payment Service Provider Identifier eg: FonePay, NepalPay
        /// </summary>
        public string TransactionType { get; set; }
        /// <summary>
        /// TransactionAmount is the total Amount that is to be paid, This amount is in Paisa, Need to convert if needed
        /// </summary>
        public decimal TransactionAmount { get; set; }
        /// <summary>
        /// This holds the status of the transaction whether the transaction is successful or failed
        /// </summary>
        public bool TransactionStatus { get; set; }
        /// <summary>
        /// ResponseMessage message returned by POS Terminal Device regarding the transaction
        /// </summary>
        public string ResponseMessage { get; set; }
        /// <summary>
        /// TransactionResponse holds the overall JSON response sent to Billing System
        /// </summary>
        public string TransactionResponse { get; set; }
        /// <summary>
        /// TransactionDate refers to the Date when Transaction is done
        /// </summary>
        public string TransactionDate { get; set; }
        /// <summary>
        /// TransactionTime holds the time when Transaction is done
        /// </summary>
        public string TransactionTime { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/shared/fewa-pay/dtos/fewa-pay-transaction-log.dto.ts
import { ENUM_POS_ResponseStatus } from "../../shared-enums";
import { FewaPayTransactionDetailsDTO } from "./fewa-pay-transaction-response.dto";

/**
 * @summary This dto class is to represent the response received from Fewa POS terminal
 */
export class FewaPayTransactionLogDTO {
  PatientId: number = 0;
  ModuleName: string = "";
  TransactionId: string = "";
  TransactionType: string = "";
  TransactionAmount: number = 0;
  TransactionStatus: boolean = false;
  ResponseMessage: string = "";
  TransactionResponse: string = "";
  TransactionDate: string | null = "";
  TransactionTime: string | null = "";

  constructor(response: FewaPayTransactionDetailsDTO, patientId: number, moduleName: string) {
    this.TransformFewaPayResponse(response, patientId, moduleName);
  }

  /**
   *
   * @param response It is the response received from FewaPay POS terminal
   * @param patientId It is the patient for whom the payment is being done
   * @returns FewaPayTransactionLogDTO
   */
  TransformFewaPayResponse(response: FewaPayTransactionDetailsDTO, patientId: number, moduleName: string): FewaPayTransactionLogDTO {
    this.PatientId = patientId;
    this.ModuleName = moduleName;
    this.TransactionId = response.verifyTransId;
    this.TransactionType = response.transactionType;
    this.TransactionAmount = response.transactionAmount ? +response.transactionAmount : 0;
    this.TransactionStatus = response.resultCode === ENUM_POS_ResponseStatus.Success ? true : false;
    this.ResponseMessage = response.message;
    this.TransactionDate = response.transactionDate;
    this.TransactionTime = response.transactionTime;
    this.TransactionResponse = JSON.stringify(response);

    return this;
  }

}
